src/GraphQL.Tests/Errors/ErrorParserTests.cs

```diff
@@ -0,0 +1,290 @@
+            error.AddLocation(5, 6);
+            error.AddLocation(7, 8);
+
+            var parsed = new ErrorParser().Parse(error);
+            parsed.Message.ShouldBe(error.Message);
+            parsed.Locations.ShouldBe(error.Locations);
+            parsed.Path.ShouldNotBeNull();
+            parsed.Path.ShouldBe(error.Path);
+            parsed.Extensions.ShouldNotBeNull();
+            parsed.Extensions.Count.ShouldBe(3);
+            parsed.Extensions.ShouldContainKeyAndValue("code", error.Code);
+            parsed.Extensions.ShouldContainKey("codes");
+            parsed.Extensions["codes"].ShouldBeAssignableTo<IEnumerable<string>>().ShouldBe(error.Codes);
+            parsed.Extensions.ShouldContainKey("data");
+            parsed.Extensions["data"].ShouldBeAssignableTo<IDictionary>().ShouldBe(error.Data);
+        }
+
+        [Fact]
+        public void exposeExceptions()
+        {
+            var innerException = new ArgumentNullException(null, new ArgumentOutOfRangeException());
+            var error = new ExecutionError(innerException.Message, innerException);
+
+            var parsed = new ErrorParser(true).Parse(error);
+            parsed.Message.ShouldBe(error.ToString());
+        }
+
+        [Fact]
+        public void exposeExceptions_with_real_stack_trace()
+        {
+            // generate a real stack trace to serialize
+            ExecutionError error;
+            try
+            {
+                try
+                {
+                    throw new ArgumentNullException(null, new ArgumentOutOfRangeException());
+                }
+                catch (Exception innerException)
+                {
+                    throw new ExecutionError(innerException.Message, innerException);
+                }
+            }
+            catch (ExecutionError e)
+            {
+                error = e;
+            }
+
+            var parsed = new ErrorParser(true).Parse(error);
+            parsed.Message.ShouldBe(error.ToString());
+        }
+
+        [Fact]
+        public void blank_codes_do_not_serialize()
+        {
+            var error = new ExecutionError(null)
+            {
+                Code = "",
+            };
+            error.Code.ShouldBe("");
+            error.HasCodes.ShouldBeFalse();
+
+            var parsed = new ErrorParser(true).Parse(error);
+            parsed.Extensions.ShouldBeNull();
+        }
+
+        [Fact]
+        public void inner_exception_of_type_exception_does_not_serialize_extensions()
+        {
+            var error = new ExecutionError("Test execution error", new Exception("Test exception"));
+            error.Code.ShouldBe("");
+            error.HasCodes.ShouldBeTrue();
+            error.Codes.ShouldBe(new[] { "" });
+
+            var parsed = new ErrorParser().Parse(error);
+            parsed.Extensions.ShouldBeNull();
+        }
+
+        [Fact]
+        public void codes_with_blank_code_has_undefined_behavior()
```

Comment:
This behavior should be reviewed.  Presently when the `InnerException` property is of type `Exception`, it prevents `code` and `codes` from serializing -- unless there is items in the `Data` collection.